### PR TITLE
Extend FAQ documentation to add configuration Q&A pairs

### PR DESCRIPTION
**Pull request by stichbury**
> **NOTE:** Kedro datasets are moving from `kedro.extras.datasets` to a separate `kedro-datasets` package in
> [`kedro-plugins` repository](https://github.com/kedro-org/kedro-plugins). Any changes to the dataset implementations
> should be done by opening a pull request in that repository.

## Description
In response to https://github.com/kedro-org/kedro/issues/2438 I have added links to "How to" Q&A pairs that were created for the configuration documentation.

This is all I'm doing in terms of FAQs right now. 

There are other commonly answered questions but I'm going to raise issues to get these added to documentation separately as they arise, and some may be better included in the FAQ on the [GitHub wiki](https://github.com/kedro-org/kedro/wiki) for now, since that holds odds and ends of queries that don't fit within the docs. 

Some, already known, are not docs issues such as dynamic pipelines, which highlight [Kedro's philosophical stance so will be covered in a blog post](https://github.com/kedro-org/kedro-devrel/issues/7).

### Adding fix for missing permalink symbol
[Thanks to @astrojuanlu there's a cherry-picked fix in here to add the missing symbol back to our permalinks](https://github.com/kedro-org/kedro/pull/2503/commits/0b2bf92809319986758780e24e9a62ba7234b7a8).

## Development notes
Linted, rebuilt locally, inspected docs.

## Checklist

- [x] Read the [contributing](https://github.com/kedro-org/kedro/blob/main/CONTRIBUTING.md) guidelines
- [ ] Opened this PR as a 'Draft Pull Request' if it is work-in-progress
- [ ] Updated the documentation to reflect the code changes
- [ ] Added a description of this change in the [`RELEASE.md`](https://github.com/kedro-org/kedro/blob/main/RELEASE.md) file
- [ ] Added tests to cover my changes

### CODE REVIEW

**Review by merelcht — APPROVED** [review]
LGTM! 👍

**Review by astrojuanlu — APPROVED** [review]
:shipit:

File: docs/source/faq/faq.md [hunk 1]
```diff
@@ -1,15 +1,41 @@
...
+## Configuration
+
+* [How do I change the setting for a configuration source folder](../configuration/configuration_basics.md#how-to-change-the-setting-for-a-configuration-source-folder)?
+* [How do I change the configuration source folder at run time](../configuration/configuration_basics.md#how-to-change-the-configuration-source-folder-at-runtime)?
+* [How do I specify parameters at run time](../configuration/parameters.md#how-to-specify-parameters-at-runtime)?
+* [How do I read configuration from a compressed file](../configuration/configuration_basics.md#how-to-read-configuration-from-a-compressed-file)?
+* [How do I access configuration in code](../configuration/configuration_basics.md#how-to-access-configuration-in-code)?
+* [How do I load credentials in code](../configuration/credentials.md#how-to-load-credentials-in-code)?
+* [How do I load parameters in code](../configuration/parameters.md#how-to-load-parameters-in-code)?
+* [How do I specify additional configuration environments ](../configuration/configuration_basics.md#how-to-specify-additional-configuration-environments)?
```
Comment (noklam) [inline docs/source/faq/faq.md:26]:
```suggestion
* [How do I specify additional configuration environments](../configuration/configuration_basics.md#how-to-specify-additional-configuration-environments)?
```

**Review by noklam — APPROVED** [review]
LGTM! thanks!

**Pull request merged**